packages/ui/src/views/repo/pull-request/pull-request-list-page.tsx

```diff
@@ -137,50 +159,127 @@ const PullRequestList: FC<PullRequestPageProps> = ({
-            <Spacer size={6} />
-            <ListActions.Root>
-              <ListActions.Left>
-                <SearchBox.Root
-                  width="full"
-                  className="max-w-96"
-                  value={searchInput || ''}
-                  handleChange={handleInputChange}
-                  placeholder={t('views:repos.search', 'Search')}
-                />
-              </ListActions.Left>
-              <ListActions.Right>
-                <Filters
-                  filterOptions={FILTER_OPTIONS}
-                  sortOptions={SORT_OPTIONS}
-                  filterHandlers={filterHandlers}
-                  t={t}
-                />
-                <Button variant="default" asChild>
-                  <Link to={`${spaceId ? `/${spaceId}` : ''}/repos/${repoId}/pulls/compare/`}>New pull request</Link>
-                </Button>
-              </ListActions.Right>
-            </ListActions.Root>
-            <FiltersBar
-              filterOptions={FILTER_OPTIONS}
-              sortOptions={SORT_OPTIONS}
-              sortDirections={SORT_DIRECTIONS}
-              filterHandlers={filterHandlers}
-              t={t}
-            />
-            <Spacer size={5} />
-          </>
-        )}
-        {renderListContent()}
-        <PaginationComponent totalPages={totalPages} currentPage={page} goToPage={setPage} t={t} />
-      </SandboxLayout.Content>
-    </SandboxLayout.Main>
+    <PRListFilterHandler
+      ref={filtersRef}
+      onFilterSelectionChange={onFilterSelectionChange}
+      onChange={onFilterValueChange}
+      view="dropdown"
+    >
+      <SandboxLayout.Main className="max-w-[1040px]">
+        <SandboxLayout.Content>
+          {showTopBar && (
+            <>
+              <Spacer size={2} />
+              <p className="text-24 font-medium leading-snug tracking-tight text-foreground-1">Pull Requests</p>
+              <Spacer size={6} />
+              <ListActions.Root>
+                <ListActions.Left>
+                  <SearchBox.Root
+                    width="full"
+                    className="max-w-96"
+                    value={searchInput || ''}
+                    handleChange={handleInputChange}
+                    placeholder={t('views:repos.search', 'Search')}
+                  />
+                </ListActions.Left>
+                <ListActions.Right>
+                  <PRListFilterHandler.Dropdown>
+                    {(addFilter, availableFilters, resetFilters) => (
+                      <FilterSelect
+                        options={PR_FILTER_OPTIONS.filter(option =>
+                          availableFilters.includes(option.value as PRListFiltersKeys)
+                        )}
+                        onChange={option => {
+                          addFilter(option.value as PRListFiltersKeys)
+                          setOpenedFilter(option.value as PRListFiltersKeys)
```

Comment:
`as` typecasting is dangerous. why is needed here?

Reply:
Same